HW_12_library.py

```diff
@@ -0,0 +1,96 @@
+import os
+os.system('cls')
+
+
+class Book:
+
+    def __init__(self, book_name, author, num_pages, isbn):
+        self.book_name = book_name
+        self.author = author
+        self.num_pages = num_pages
+        self.isbn = isbn
+        self.reserve_flag = False
+        self.reserve_reader = None
+        self.current_reader = None
+
+    def reserve(self, reader):
+        if not self.reserve_flag and self.current_reader != reader:
+            self.reserve_flag = True
+            self.reserve_reader = reader
+            print(f'Success reserve. {self.book_name} has been reserved by {reader.name}.')
+        elif not self.reserve_flag:
+            self.reserve_flag = True
+            self.reserve_reader = reader
+            print(f'Success reserve. {self.book_name} has been reserved by {reader.name}.')
+        else:
+            print(f'Reserve error. {self.book_name} is already reserved/received.')
```

Comment:
Better to return a boolean flag from methods like this, and process it where it was executed